treewide: Remove nRF Cloud REST (#28522)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-1.8.0.rst

See `Modem firmware compatibility matrix`_ for an overview of which modem firmware versions have been tested with this version of the |NCS|.

Use the latest version of the nRF Programmer app of `nRF Connect for Desktop`_ to update the modem firmware.

> See the ``Programming nRF91 Series DK firmware`` page for instructions.

Reviewer: ```suggestion
See the `Programming nRF91 Series DK firmware`_ page for instructions.
```

--------------

* Updated several samples and libraries.

>   For details, see ``Bluetooth mesh samples``_ and ``Bluetooth libraries``, respectively.

Reviewer: ```suggestion
  For details, see :ref:`bt_mesh_samples` and :ref:`lib_bluetooth_services`, respectively.
```

Author: I guess this is outside the scope of REST lib removal

* Added:

>   * ``nrf_cloud_rest_fota`` sample, demonstrating how to perform FOTA updates with the nRF Cloud REST API.

Reviewer: ```suggestion
  * Cellular: nRF Cloud REST FOTA sample, demonstrating how to perform FOTA updates with the nRF Cloud REST API.
```

  * Changed :c:func:`nrf_cloud_pgps_init` to limit allowable :kconfig:option:`CONFIG_NRF_CLOUD_PGPS_NUM_PREDICTIONS` to an even number, and limited :kconfig:option:`CONFIG_NRF_CLOUD_PGPS_REPLACEMENT_THRESHOLD` to this value minus 2.
  * Updated the signature of :c:func:`npgps_download_start` to accept an integer parameter specifying the PDN ID, which replaces the parameter used to specify the APN.

> * ``lib_nrf_cloud_rest`` library:

Reviewer: ```suggestion
* nRF Cloud REST library:
```
